homeassistant/components/climate/wink.py

```diff
@@ -0,0 +1,348 @@
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self.wink.current_humidity()
+
+    @property
+    def external_temperature(self):
+        """Return the current external temperature."""
+        return self.wink.current_external_temperature()
+
+    @property
+    def smart_temperature(self):
+        """
+        Return the current average temp of all remote sensor (ecobee only).
+        """
+        return self.wink.current_smart_temperature()
+
+    @property
+    def eco_target(self):
+        """Return status of eco target (Is the termostat in eco mode)."""
+        return self.wink.eco_target()
+
+    @property
+    def occupied(self):
+        """Return status of if the thermostat has detected occupancy."""
+        return self.wink.occupied()
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        if not self.wink.is_on():
+            return STATE_OFF
+        elif self.wink.current_hvac_mode() == 'cool_only':
+            return STATE_COOL
+        elif self.wink.current_hvac_mode() == 'heat_only':
+            return STATE_HEAT
+        elif self.wink.current_hvac_mode() == 'aux':
+            return STATE_HEAT
+        elif self.wink.current_hvac_mode() == 'auto':
+            return STATE_AUTO
+        else:
+            return STATE_UNKNOWN
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        #Should this be a percentage or float?
+        if self.wink.current_humidifier_mode() == 'on':
+            return self.wink.current_humidifier_set_point()
+        elif self.wink.current_dehumidifier_mode() == 'on':
+            return self.wink.current_dehumidifier_set_point()
+        else:
+            return None
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self.current_operation != STATE_AUTO and not self.is_away_mode_on:
+            if self.current_operation == STATE_COOL:
+                return self.wink.current_max_set_point()
+            elif self.current_operation == STATE_HEAT:
+                return self.wink.current_min_set_point()
+            else:
+                return None
+        else:
+            return None
+
+    @property
+    def target_temperature_low(self):
+        """Return the lower bound temperature we try to reach."""
+        if self.current_operation == STATE_AUTO:
+            return self.wink.current_min_set_point()
+        return None
+
+    @property
+    def target_temperature_high(self):
+        """Return the higher bound temperature we try to reach."""
+        if self.current_operation == STATE_AUTO:
+            return self.wink.current_max_set_point()
+        return None
+
+    @property
+    def is_away_mode_on(self):
+        """Return if away mode is on."""
+        return self.wink.away()
```

Comment:
Question about wink in general. I see a lot of function calls inside `@property` methods. For performance reasons, Home Assistant does not allow fetching any dat from the cloud inside properties. So can you confirm that no requests are made to the cloud inside Wink Entity `@property` methods?